framework/PostProcessors/ExternalPostProcessor.py

```diff
@@ -90,52 +91,44 @@ def inputToInternal(self, currentInput):
     if type(currentInput) == dict and 'targets' in currentInput.keys():
       return
 
-    if type(currentInput) != list:
-      currentInput = [currentInput]
-
-    inputDict = {'targets':{}, 'metadata':{}}
-    metadata = []
-    for item in currentInput:
-      inType = None
-      if hasattr(item, 'type'):
-        inType = item.type
-      elif type(item) in [list]:
-        inType = "list"
-
-      if isinstance(item,Files.File):
-        if currentInput.subtype == 'csv':
-          self.raiseAWarning(self, 'Input type ' + inType + ' not yet implemented. I am going to skip it.')
-      elif inType == 'HDF5':
-        # TODO
-        self.raiseAWarning(self, 'Input type ' + inType + ' not yet implemented. I am going to skip it.')
-      elif inType == 'PointSet':
-        for param in item.getParaKeys('input'):
-          inputDict['targets'][param] = item.getParam('input', param)
-        for param in item.getParaKeys('output'):
-          inputDict['targets'][param] = item.getParam('output', param)
-        metadata.append(item.getAllMetadata())
-      elif inType =='HistorySet':
-        outs, ins = item.getOutParametersValues(nodeId = 'ending'), item.getInpParametersValues(nodeId = 'ending')
-        for param in item.getParaKeys('output'):
-          inputDict['targets'][param] = [value[param] for value in outs.values()]
-        for param in item.getParaKeys('input'):
-          inputDict['targets'][param] =  [value[param] for value in ins.values()]
-        metadata.append(item.getAllMetadata())
-      elif inType != 'list':
-        self.raiseAWarning(self, 'Input type ' + type(item).__name__ + ' not recognized. I am going to skip it.')
-
-      # Not sure if we need it, but keep a copy of every inputs metadata
-      inputDict['metadata'] = metadata
+    if type(currentInput) == list:
+      if len(currentInput) != 1:
+        self.raiseAnError(IOError, "The postprocessor ", self.name, "only allows one input DataObjects,"
+              + " but multiple inputs are provided!")
+      else:
+        currentInput = currentInput[-1]
+    if hasattr(currentInput, 'type'):
+      inType = currentInput.type
+    else:
+      self.raiseAnError(IOError, "Input type ", type(currentInput).__name__, ' is not recognized!')
```

Comment:
In raven, all the objects have the ```.type``` attribute.

Reply:
Is it true? we used to check this.

Reply:
fixed